### DISCUSSION TITLE
Can't connect - any advice?

### DISCUSSION THREAD

**Discussion by essessessess** [artisan-roaster-scope/artisan#840] [General]
Hi there, 

I'm struggling to connect Artisan to my Hottop. I've tried Artisan 2.4.6 on a 2021 M1 macbook pro running Monterey, with a USB-2 to USB-C cable (no adapter). I've also tried Artisan 2.4.2 on an older 2013 macbook pro, running High Sierra 10.13. Neither setup worked. I've read elsewhere that v2.1.2 has been used successfully with a 2013 macbook pro. However the DMGs for 2.1.2 are no longer available, just a .zip and a tar.gz, and no installation instructions for these. Would anyone kindly offer some advice for next steps?

I've contacted Hottop repeatedly, and while prompt and friendly in their response, they just say they only have experience with PCs.

**Comment by MAKOMO** [comment]
There is no known issue in v2.4.6 regarding communication to a Hottop. There are some known issues with non-USBC adapters (or wires) to communicate at all on the latest macOS versions (Monterey). Nobody knows if this is intensional by Apple to rule those non-Apple adapters out or if this is just a bug.

**Comment by poundy** [comment]
Get a PC to try this on. 
Also, there's no reason to try older versions of Artisan - they are no longer supported, and as Marko says there's no good reaason not to use the current version as there's no known issue... I'm sorry, but that puts it firmly back to your local PC and conneecting cables/modules.

**Comment by essessessess** [comment]
Thanks for the advice. I'm afraid I still can't connect, using the most recent version of Artisan, having tried two Macs and two cables. I don't have access to a PC unfortunately. 

I'm using a brand new KN-8828B-2K+. Are there any known connectivity issues with this unit?

**Comment by cparker333** [comment]
I had the same problem connecting an upgraded Hottop (KN-8288B to KN-8288B-K+) to a MacBook running Big Sur (11.6).  I tried connecting with a USB-A to UCB-C cable (no adaptor) and could not get the two to communicate.  Finally broke down and got a $20, Apple-branded USB-to-USB-C adaptor and was able to get Artisan to communicate with the Hottop immediately.  The only thing I changed was to select the second port option offered (not the FTDI default you usually see).  Other adapters may work, but Apple seems to nerf some of them.

**Comment by essessessess** [comment]
Thanks for the tip, I will try that.

>

**Reply by cparker333** [reply]
Just to add, I had to specifically plug it in to the port on the left side of the laptop.  The right side didn't work for some reason.

**Comment by staynate** [comment]
Hey! I am having this same issue with each version of Artisan that I download. I am using a MacBook Pro (2014) that is currently running macOS Big Sur (version 11.6.7). The roaster I have is a Hottop KN-8828b-2k+. I followed the steps for configuring my machine with Artisan, but nothing happened. Also, I am using USB-A going into my computer.

**Reply by poundy** [reply]
Can you try from a PC? 
There's a theme with many USB connected roasters and Macs, basically you need to have a "recognised" USB cable, meaning one that is Mac "certified" or a genuine Apple one, not your everyday cheapie that often comes with typical devices including roasters... If you can try from a PC, that proves the roaster is working, and then it can confirm the issue is the mac and cable incompatibility

**Reply by staynate** [reply]
Alright. People of got to stop acting like "try from a PC" is an actual solution. I have a Mac, and not going to just go buy a PC.

I used the advice from @cparker333, and plugged it in on the left side of my MacBook Pro (2014). There is one USB port on that side- to be clear. Then, while configuring my machine, I had to select the second port option "FT230X Basic UART". The option which says "/dev/cu.usbserial-DA01POGR" did not connect my Hottop roaster to my computer.

![IMG_0435](https://user-images.githubusercontent.com/108427907/176582917-8aa08484-bc3b-4768-8a64-e1466c434822.JPG)

After connecting it like that i was able to complete a roast with no problem and am about to do another.

Maybe this is actual solution for people with older Macs?

**Reply by poundy** [reply]
Well, lets get clear here - my question about trying a PC was just that - and to quote, "can you try it on a PC?" and didn't try to claim this would make it work or not, it'd simply give us extra data points.  And, since I'm *that* PC guy, who hasn't touched a Mac since the Mac's in Uni, when there was no Linux just Unix (yes a long time ago), I am not going to be able to use any experience here to assist :) 

So. Lets also point out that a USB port should be a USB port :) You plug a USB cable in an open USB port in the same machine, left side, right side, back side or whatever, it should always behave the same.  So as an additional test, what happens when you follow your now confirmed working steps, only using a different physical USB socket, does that work? If not, there's something special about this age Mac and how it's USB ports are handled, and it would appear that the only valid port to use these FTDI-based devices on - so yes, that would be an actual solution.  (It also confirms, this isn't Artisan, it's the MacOS+hardware...)
